homeassistant/components/shelly/climate.py

```diff
@@ -98,20 +98,25 @@ async def async_restore_climate_entities(
         ent_reg, config_entry.entry_id
     )
 
+    restore_entries = []
     for entry in entries:
 
         if entry.domain != CLIMATE_DOMAIN:
             continue
 
         _LOGGER.debug("Setup sleeping climate device %s", wrapper.name)
         _LOGGER.debug("Found entry %s [%s]", entry.original_name, entry.domain)
-        async_add_entities([BlockSleepingClimate(wrapper, None, None, entry)])
+
+        restore_entries.append(BlockSleepingClimate(wrapper, None, None, entry))
+
+    if restore_entries:
+        async_add_entities(restore_entries)
```

Comment:
As far as I understand we can have only one entry for each `config_entry`?
```suggestion
        async_add_entities([BlockSleepingClimate(wrapper, None, None, entry)])
        return
```

Reply:
Shay, I was explicitly asked to first collect all entities and then add them ;-)

Reply:
Because when @MartinHjelmare reviewed your previous code he assumed you can have more than one entity. Can we have more than one climate entity for a config entry?

Reply:
I can have more entries:

```python
for entry in entries:
```

This is why I was correctly asked to collect all entities.

Simone

Reply:
This is where you setup the climate entities when not restoring entities (device not sleeping):
https://github.com/home-assistant/core/blob/1a4ac912eaca56da0408ec494317a457c82525c8/homeassistant/components/shelly/climate.py#L83-L85

How would you have more than one climate entity per device if it was not setup when the device was not sleeping?

Reply:
Reverted

Reply:
This should not be reverted, you are missing a return there